### PR TITLE
Multiple loss functions

### LINKED ISSUE #526: Support multiple loss functions
## Subject of the feature

What are you trying to do and how would you want to do it differently?

I'd like to use multiple loss functions together, e.g. dice loss + cross-entropy.
But it is not possible for the moment I guess.

If the feature request is approved, would you be willing to submit a PR? _(Help can be
provided if you need assistance submitting a PR)_

Yes

### PR DESCRIPTION
**Pull request by acasamitjana**
# Description

Support the specification of >1 loss functions for each loss type "image", "label", "regularization". I'm not sure it make sense to use multiple loss functions for regularization but anyway it's done.

A config file example is:
```
loss:
  "image"
      - "name": "ssd"
         "weight": 0.5
      - "name": "lncc"
         "weight": 0.5
  "label"
       "name": "cross-entropy"
       "weight": 1.0
  "regularization":
        "name": "bending"
        "weight": 0.1
...
```
      
Fixes #526 

## Type of change

What types of changes does your code introduce to DeepReg?

_Please check the boxes that apply after submitting the pull request._

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation Update (fix or improvement on the documentation)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Other (if none of the other choices apply)

## Checklist

_Please check the boxes that apply after submitting the pull request._

_If you're unsure about any of them, don't hesitate to ask. We're here to help! This is
simply a reminder of what we are going to look for before merging your code._

- [X] I have
      [installed pre-commit](https://deepreg.readthedocs.io/en/latest/contributing/setup.html)
      using `pre-commit install` and formatted all changed files. If you are not
      certain, run `pre-commit run --all-files`.
- [X] My commits' message styles matches
      [our requested structure](https://deepreg.readthedocs.io/en/latest/contributing/commit.html),
      e.g. `Issue #<issue number>: detailed message`.
- [X] I have updated the
      [change log file](https://github.com/DeepRegNet/DeepReg/blob/main/CHANGELOG.md)
      regarding my changes.
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by acasamitjana** [comment]
Hi, @mathpluscode @YipengHu can you take a look at that? It's related to one of your previous comments.

**Comment by acasamitjana** [comment]
Hi, @mathpluscode @NMontanaBrown can you take a look at that? Really, it's not much. Now, the loss under each type (e.g, "image") can be a list(DictWrapper()) or a DictWrapper() for >1 or 1 losses, respectively.
To do that, I'm converting the later to a list in the code. If you know a better way to do it, I'm all ears!

**Review by NMontanaBrown — APPROVED** [review]
Nice work, thanks @acasamitjana :)

**Review by YipengHu — APPROVED** [review]
@acasamitjana multiple losses on ddf will be very useful for practical applications, such as known correspondence - one user can add readily with register

**Review by mathpluscode — CHANGES_REQUESTED** [review]
Related documents need to be updated.

File: deepreg/model/network.py [hunk 1]
```diff
@@ -186,39 +188,47 @@ def _build_loss(self, name: str, inputs_dict: dict):
             )
             return
 
-        loss_config = self.config["loss"][name]
+        loss_list = self.config["loss"][name]
```
Comment (mathpluscode) [inline deepreg/model/network.py:191]: I may prefer the var `loss_configs` or `loss_config_list`

File: test/unit/test_config_v011.py [hunk 2]
```diff
@@ -85,55 +85,78 @@ def test_parse_loss():
...
-        },
-    }
-    got = parse_image_loss(loss_config=loss_config)
-    assert got == expected
-
-
-class TestParseLabelLoss:
-    def test_label_multi_scale(self):
+class TestParseImageLoss:
+    def test_parse_old_loss(self):
```
Comment (mathpluscode) [inline test/unit/test_config_v011.py:89]: the test name should be `test_parse_single_loss`

**Comment by mathpluscode** [comment]
@acasamitjana I think the most important is to update the docs, otherwise nice work!

**Comment by acasamitjana** [comment]
@mathpluscode what do you think about the docs update? I've added some text + example in "Multiple Loss" (now reads "Composite Loss", which I think is more accurate). Feel free to request any changes.

**Review by mathpluscode — APPROVED** [review]
